apps/workers/workers/videoWorker.ts

```diff
@@ -24,7 +24,10 @@ import {
 import { withTimeout } from "../utils";
 import { getBookmarkDetails, updateAsset } from "../workerUtils";
 
-const TMP_FOLDER = path.join(os.tmpdir(), "video_downloads");
+// Use the configured data directory if available, fallback to tmp directory otherwise
+const TMP_FOLDER = serverConfig.dataDir
+  ? path.join(serverConfig.dataDir, "video_downloads")
+  : path.join(os.tmpdir(), "video_downloads");

```

Comment:
can we name this `tmp_video_downloads` now that it's going to be in a persistent directory? Also, can we clean everything inside it on worker startup? Otherwise we might be leaking data there.